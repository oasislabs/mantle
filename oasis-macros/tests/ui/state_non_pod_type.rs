oasis_std::contract! {

#[derive(Contract)]
pub struct NonPOD(*const u8);

impl NonPOD {
    pub fn new(ctx: &Context) -> Self {
        Self(std::ptr::null())
    }
}

}

fn main() {}
