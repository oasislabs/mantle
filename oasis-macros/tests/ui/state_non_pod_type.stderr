error: Contract field must be a POD type.
 --> $DIR/state_non_pod_type.rs:4:19
  |
4 | pub struct NonPOD(*const u8);
  |                   ^

error: aborting due to previous error

